src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.MovieDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    private static final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("There is something wrong with saving Movie "
+                    + movie + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        Session session = sessionFactory.openSession();
+        return Optional.ofNullable(session.get(Movie.class, id));
```

Comment:
Common mistakes:
Use try with finally, where you use transaction and try with resources with Read operations.